include/circt/Analysis/DependenceAnalysis.h

```diff
@@ -66,6 +66,9 @@ struct MemoryDependenceAnalysis {
   // Returns the dependences, if any, that the given Operation depends on.
   ArrayRef<MemoryDependence> getDependences(Operation *);
 
+  // Replaces the dependences, if any, from the source op to the destination op.
+  void replaceOp(Operation *source, Operation *destination);
```

Comment:
Could that be `(Operation *oldOp, Operation *newOp)` or something similar? I think "source" and "destination" are potentially confusing here, considering that this analysis is about dependences that themselves have a source and destination.